### DISCUSSION TITLE
How can I get Zilean to return results?

### DISCUSSION THREAD

**Discussion by ojread** [I-am-PUID-0/DMB#76] [Q&A] [answered]
I think I have everything working apart from Zilean. Any searches it performs return no results. Testing with the Swagger frontend, the healthcheck and IMDB endpoints work but the DMM ones return nothing. Is there anything I can try?

```
dmb  | Nov 15, 2024 06:54:48 - INFO - Zilean subprocess:  | "Zilean.ApiService.Features.Dmm.DmmEndpoints.DmmFilteredInstance" | Performing filtered search for DmmFilteredRequest { Query: "Deadpool & Wolverine", Season: null, Episode: null,
dmb  | Nov 15, 2024 06:54:48 - INFO - Zilean subprocess: Year: null, Language: null, Resolution: null, ImdbId: null }
dmb  | Nov 15, 2024 06:54:48 - INFO - Zurg w/ RealDebrid subprocess: | manager | Detected changes! Refreshing 0 torrents
dmb  | Nov 15, 2024 06:54:48 - INFO - riven_frontend subprocess: Listening on 0.0.0.0:3000
dmb  | Nov 15, 2024 06:54:48 - INFO - Zilean subprocess:  | "Zilean.ApiService.Features.Dmm.DmmEndpoints.DmmFilteredInstance" | Filtered search for "Deadpool & Wolverine" returned 0 results
dmb  | Nov 15, 2024 06:54:48 - INFO - riven_backend subprocess: | 👻 SCRAPER   | shared._parse_results - Processing 0 results for Deadpool & Wolverine
dmb  | Nov 15, 2024 06:54:48 - INFO - riven_backend subprocess: | 🤷‍ NOT_FOUND | __init__.run - Scraping returned no good results for Deadpool & Wolverine
```

**Comment by I-am-PUID-0** [comment]
Unfortunately, I'm unsure why Zilean is not returning results, and I'm unable to replicate the issue, as both DMM and IMDB return results on my system.

Has Zilean completed the initial caching process after the first startup? 

DMM:
```
[
  {
    "raw_title": "X-Men Origins Wolverine 2009 1080p BrRip x264 YIFY mp4",
    "parsed_title": "X-Men Origins Wolverine",
    "normalized_title": "xmen origins wolverine",
    "trash": false,
    "year": 2009,
    "resolution": "1080p",
    "seasons": [],
    "episodes": [],
    "complete": false,
    "volumes": [],
    "languages": [],
    "quality": "BRRip",
    "hdr": [],
    "codec": "avc",
    "audio": [],
    "channels": [],
    "dubbed": false,
    "subbed": false,
    "date": null,
    "group": null,
    "edition": null,
    "bit_depth": null,
    "bitrate": null,
    "network": null,
    "extended": false,
    "converted": false,
    "hardcoded": false,
    "region": null,
    "ppv": false,
    "_3d": false,
    "site": null,
    "size": "1450816651",
    "proper": false,
    "repack": false,
    "retail": false,
    "upscaled": false,
    "remastered": false,
    "unrated": false,
    "documentary": false,
    "episode_code": null,
    "country": null,
    "container": "mp4",
    "extension": null,
    "torrent": false,
    "category": "movie",
    "imdb_id": "tt0458525",
    "imdb": null,
    "info_hash": "6b586abefdb2efaf2f125103bde1e7665742f059"
  },
  ```
  
  IMDB:
  ```
  [
  {
    "title": "Deadpool & Wolverine",
    "imdbId": "tt6263850",
    "year": 2024,
    "score": 1,
    "category": "movie"
  }
]
```

**Reply by ojread** [reply]
Thank you for your time! Yes, Zilean has completed the initial caching process and I got the DMMSyncJob Complete message. I've tried running this on two different servers and have the same issue on both. I'm not sure what else to try.

**Reply by ojread** [reply]
I tried running with ZILEAN_UPDATE=true and it seems get updated but then it complains that the installed version of .NET is wrong - the new version needs v9.

**Reply by I-am-PUID-0** [reply]
Yep, Zilean just updated to dotnet 9. My dev machine is offline at the moment, but once it's up, I'll push an update to fix the dotnet version.

**Reply by I-am-PUID-0** [reply]
Should be resolved w/ [Version [5.4.4]](https://github.com/I-am-PUID-0/DMB/commit/704fba3498892334be9a73bf23b087dde19e894e)

**Reply by ojread** [reply]
Thanks a lot, I'm still not getting any results from the latest Zilean though. It's downloaded all the hashlists and everything else seems to be working execpt this one step where it only returns an empty results list. This is on a normal, freshly installed Debian server so I'm struggling to think what could be different for me.

**Reply by ojread** [reply]
FYI the Zilean project on its own does the same so it's obviously a me problem.

**Reply by ojread** [reply, marked as answer]
Success! I found the env var Zilean__Dmm__ImportBatched=true which causes the import to take a lot longer but at least it works and I can search now.
